lib/spack/spack/solver/asp.py

```diff
@@ -3230,35 +3179,49 @@ def define_runtime_constraints(self):
         recorder = RuntimePropertyRecorder(self)
 
         for compiler in self.possible_compilers:
-            compiler_with_different_cls_names = {
-                "oneapi": "intel-oneapi-compilers",
-                "clang": "llvm",
-            }
-            compiler_cls_name = compiler_with_different_cls_names.get(
-                compiler.spec.name, compiler.spec.name
-            )
             try:
-                compiler_cls = spack.repo.PATH.get_pkg_class(compiler_cls_name)
-                if hasattr(compiler_cls, "runtime_constraints"):
-                    compiler_cls.runtime_constraints(spec=compiler.spec, pkg=recorder)
+                compiler_cls = spack.repo.PATH.get_pkg_class(compiler.name)
             except spack.repo.UnknownPackageError:
                 pass
+            else:
+                if hasattr(compiler_cls, "runtime_constraints"):
+                    compiler_cls.runtime_constraints(spec=compiler, pkg=recorder)
+                # Inject default flags for compilers
+                recorder("*").default_flags(compiler)
+
+            # FIXME (compiler as nodes): think of using isinstance(compiler_cls, WrappedCompiler)
+            # Add a dependency on the compiler wrapper
```

Comment:
Couldn't wrapped compiler classes inject this instead? i.e. WrappedCompiler is the base class you use to get this, and *it* defines `runtime_constriants` to inject the wrapper. Not necessary for this PR -- there is already a note about the generality here.

I think this is related to decoupling the repo as well.  You should probably be able to build a repo that doesn't have a wrapper.

Reply:
I agree with making this more general in a following PR.